Bottom: bdad6ef078aa70b4779c4b5844ac469dbb3267d6
Top:    14bddf38066dfd6039a232bc0916d2e902aa8369
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 15:05:49 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 8c78fea..113a7ba 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -307,6 +307,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
